resource/ui/ts/models/node_status.ts

```diff
@@ -0,0 +1,126 @@
+        import promise = _mithril.MithrilPromise;
+
+        /**
+        */
+        export interface IAddress {
+            network: string;
+            address: string;
+        }
+
+        /**
+        */
+        export interface INodeDescription {
+            node_id: number;
+            address: IAddress;
+            attrs: any;
+        }
+
+        /**
+        */
+        export interface INodeStatus {
+            desc: INodeDescription;
+            store_ids: number[];
+            range_count: number;
+            started_at: number;
+            updated_at: number;
+            stats: Models.Stats.IMVCCStats;
+            leader_range_count: number;
+            replicated_range_count: number;
+            available_range_count: number;
+        }
+
+        export interface INodeStatusResponseSet {
+            d: INodeStatus[]
+        }
+
+        export interface INodeStatusListMap {
+            [nodeId: number]: INodeStatus[]
+        }
+
+        export interface INodeDescriptionMap {
+            [nodeId: number]: INodeDescription
+        }
+
+        export interface INodeStatusMap {
+            [nodeId: number]: INodeStatus
+        }
+
+        export class Nodes {
+            private _dataLimit = 100000;
+            private _dataPrunedSize = 90000;
+            private _data: _mithril.MithrilProperty<INodeStatusListMap> = m.prop(<INodeStatusListMap> {});
+
+            public desc: _mithril.MithrilProperty<INodeDescriptionMap> = m.prop(<INodeDescriptionMap> {});
+            public statuses: _mithril.MithrilProperty<INodeStatusMap> = m.prop(<INodeStatusMap> {});
+
+            QueryAll(): _mithril.MithrilPromise<INodeStatusResponseSet> {
+                var url = "/_status/nodes/";
+                return m.request({ url: url, method: "GET", extract: nonJsonErrors })
+                    .then((results: INodeStatusResponseSet) => {
+                        results.d.forEach((status) => {
+                            if (this._data()[status.desc.node_id] == null) {
+                                this._data()[status.desc.node_id] = <Models.NodeStatus.INodeStatus[]>[];
+                            }
+                            this._data()[status.desc.node_id].push(status);
+                            this.statuses()[status.desc.node_id] = status;
+                        });
+                        this._pruneOldEntries();
+                        this._updateDescriptions();
+                        return results;
+                    });
+            }
+
+            QueryOne(nodeId:string): _mithril.MithrilPromise<INodeStatus> {
+                var url = "/_status/nodes/" + nodeId;
+                return m.request({ url: url, method: "GET", extract: nonJsonErrors })
+                    .then((status: INodeStatus) => {
+                        if (this._data()[status.desc.node_id] == null) {
+                            this._data()[status.desc.node_id] = <Models.NodeStatus.INodeStatus[]>[];
```

Comment:
I don't think you need to cast here, the statement already has type information from the... is it still lvalue in javascript?  The variable being assigned has a type.